src/lsm6.ts

```diff
@@ -0,0 +1,305 @@
+GYRO_OUTPUT_SCALE_FACTOR.set(GyroScale.SCALE_125_DPS, 4.375);
+GYRO_OUTPUT_SCALE_FACTOR.set(GyroScale.SCALE_250_DPS, 8.75);
+GYRO_OUTPUT_SCALE_FACTOR.set(GyroScale.SCALE_500_DPS, 17.5);
+GYRO_OUTPUT_SCALE_FACTOR.set(GyroScale.SCALE_1000_DPS, 35);
+GYRO_OUTPUT_SCALE_FACTOR.set(GyroScale.SCALE_2000_DPS, 70);
+
+export default class LSM6 {
+    private _accel: Vector3 = { x: 0, y: 0, z: 0 };
+    private _gyro: Vector3 = { x: 0, y: 0, z: 0 };
+
+    private _i2cBus: I2CPromisifiedBus;
+    private _i2cAddress: number;
+
+    private _isReady: boolean = false;
+
+    private _accelerometerScale: AccelerometerScale = AccelerometerScale.SCALE_2G;
+    private _gyroScale: GyroScale = GyroScale.SCALE_250_DPS;
+
+    constructor(bus: I2CPromisifiedBus, address: number) {
+        this._i2cAddress = address;
+        this._i2cBus = bus;
+
+    }
+
+    public init(): Promise<void> {
+        return this._readByte(RegAddr.WHO_AM_I)
+        .then(whoami => {
+            if (whoami !== DS33_WHO_ID) {
+                console.log("[IMU] Invalid WHO_AM_I response");
+            }
+            else {
+                console.log("[IMU] Identified as LSM6DS33");
+                this._isReady = true;
+            }
+        });
+    }
+
+    public get accelerationG(): Vector3 {
+        return this._accel;
+    }
+
+    public get gyroDPS(): Vector3 {
+        return this._gyro;
+    }
+
+    public async enableDefault(): Promise<void> {
+        if (!this._isReady) {
+            return;
+        }
+
+        // Accelerometer
+        await this.setAccelerometerScale(AccelerometerScale.SCALE_2G);
+
+        // Gyro
+        await this.setGyroScale(GyroScale.SCALE_250_DPS);
+
+        // Common
+        await this._writeByte(RegAddr.CTRL3_C, 0x04);
+    }
+
+    public async setAccelerometerScale(scale: AccelerometerScale): Promise<void> {
+        if (!this._isReady) {
+            return;
+        }
+
+        const controlByte = ACCEL_SCALE_CTRL_BYTE.get(scale);
+        await this._writeByte(RegAddr.CTRL1_XL, controlByte);
+        this._accelerometerScale = scale;
+
+        console.log(`[IMU] Accelerometer Scale: ${scale}`);
+    }
+
+    public async setGyroScale(scale: GyroScale): Promise<void> {
+        if (!this._isReady) {
+            return;
+        }
+
+        const controlByte = GYRO_SCALE_CTRL_BYTE.get(scale);
+        await this._writeByte(RegAddr.CTRL2_G, controlByte);
+        this._gyroScale = scale;
+
+        console.log(`[IMU] Gyro Scale: ${scale}`);
+    }
+
+    /**
+     * Read the current accelerometer values and update our internal data
+     *
+     * The values obtained from the raw I2C read operations are the
+     * 16-bit twos-complement raw IMU values. They will need to be
+     * multiplied by the sensitivity values to obtain a reading in mG.
+     *
+     * The final values that get stored are in G
+     *
+     * Datasheet: https://www.pololu.com/file/0J1087/LSM6DS33.pdf
```

Comment:
This would be useful at the top of the file as a comment header to know where to go to get all data specs.